lib/lightning/work_orders.ex

```diff
@@ -630,46 +629,24 @@ defmodule Lightning.WorkOrders do
     )
   end
 
-  defp get_first_runs_for_retry(orders_ids) do
-    run_numbers_query =
-      from(r in Run,
-        where: r.work_order_id in ^orders_ids,
-        select: %{
-          id: r.id,
-          row_num:
-            row_number()
-            |> over(
-              partition_by: r.work_order_id,
-              order_by: coalesce(r.started_at, r.inserted_at)
-            )
-        }
-      )
-
-    from(r in Run,
-      join: rn in subquery(run_numbers_query),
-      on: r.id == rn.id,
-      join: wo in assoc(r, :work_order),
-      join: dc in assoc(r, :dataclip),
-      where: rn.row_num == 1 and is_nil(dc.wiped_at),
-      select: r.id,
-      order_by: [asc: wo.inserted_at]
-    )
-    |> Repo.all()
-  end
-
-  defp preload_runs_for_retry(runs_ids) do
-    from(r in Run,
-      where: r.id in ^runs_ids,
-      order_by: [asc: r.inserted_at],
+  def workorders_with_first_runs(workorders_ids) do
+    from(w in WorkOrder,
+      where: w.id in ^workorders_ids,
```

Comment:
for the asynchronous execution (on the background job) please make sure the run dataclip was not wiped out.

Reply:
Thanks for the reminder, this is fixed.